test/service_test.cc

```diff
@@ -111,6 +113,44 @@ class ServiceTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
   int grpc_server_port_{0};
 };
 
+class ServiceTestWithParameterizedConstructor : public ServiceTest {
+public:
+  void SetUp() override {
+    auto logging_context = std::make_unique<Envoy::Logger::Context>(
+        spdlog::level::info, "%L %n [%g:%#] %v", log_lock_, false);
+    service_ = std::make_unique<ServiceImpl>(std::move(logging_context));
+    grpc::ServerBuilder builder;
+    loopback_address_ = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+
+    builder.AddListeningPort(fmt::format("{}:0", loopback_address_),
+                             grpc::InsecureServerCredentials(), &grpc_server_port_);
+    builder.RegisterService(service_.get());
+    server_ = builder.BuildAndStart();
+    setupGrpcClient();
+    setBasicRequestOptions();
+  }
+
+  Envoy::Thread::MutexBasicLockable log_lock_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ServiceTestWithParameterizedConstructor,
+                         ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                         Envoy::TestUtility::ipTestParamsToString);
+
```

Comment:
Recognize that this is copy-paste of initial code, but I'm going to review the new code to match better style guide conventions.

First: this comment should be superfluous with a good test name.. We can remove that, and then change the test name.

Test names should ideally read like a sentence explaining what the test case is.  I'll defer to you on what makes sense for a name but typically it's something like:
ConstructorWithLoggingParameterCanRespondToRequests

Reply:
It looks like we typically match test suite name to the class name. ConstructorWithLoggingParameterCanRespondToRequests sounds like a good test name in TEST_P:
http://google3/third_party/googletest/googletest/include/gtest/gtest-param-test.h?l=416&rcl=317696457

Reply:
Oh, I guess you mean the line 141. Now I get it.